An example PR demonstrating the new Operations API (throwaway)

graham-chainlink opened the pull request:
Showcasing a sequence performing 3 operations
 - deploy link token (with no retry policy)
 - grant mint role
 - mint token  (with modified input on retry)

Logs: 
```
=== RUN   TestLinkChangeset
    logger.go:146: 2025-03-19T16:37:52.210+1100	INFO	operations/execute.go:75	Executing sequence	{"version": "unset@unset", "id": "link-example-sequence", "version": "1.0.0", "description": "Deploy LINK token contract, grants mint and mints some amount to same address"}
    logger.go:146: 2025-03-19T16:37:52.210+1100	INFO	operations/operation.go:68	Executing operation	{"version": "unset@unset", "id": "deploy-link-token", "version": "1.0.0", "description": "Deploy Contract Operation"}
    logger.go:146: 2025-03-19T16:37:52.213+1100	INFO	deployment/helpers.go:173	Deployed contract	{"version": "unset@unset", "Contract": "LinkToken 1.0.0", "addr": "0x4Aa6Af0356cBCCFD16E5dD0b6a88e2A2845F032B", "chain": " (909606746561742123)"}
    logger.go:146: 2025-03-19T16:37:52.213+1100	INFO	operations/operation.go:68	Executing operation	{"version": "unset@unset", "id": "grant-mint-role", "version": "1.0.0", "description": "Grant Mint Role Operation"}
    logger.go:146: 2025-03-19T16:37:52.215+1100	INFO	operations/operation.go:68	Executing operation	{"version": "unset@unset", "id": "mint-link-token", "version": "1.0.0", "description": "Mint LINK Operation"}
--- PASS: TestLinkChangeset (0.01s)
````

Build upon the PRs below
- https://github.com/smartcontractkit/chainlink/pull/16756
- https://github.com/smartcontractkit/chainlink/pull/16801
- https://github.com/smartcontractkit/chainlink/pull/16833

Future enhancements
- Caching and Idempotency behaviour for repeat runs + CI retries
- Saving of reports to json file ?

jkongie commented on deployment/common/changeset/example/operations/link_example.go:
```diff
@@ -0,0 +1,56 @@
+	Address common.Address
+}
+
+type EthereumDeps struct {
```
In the future we should call this EVM deps

graham-chainlink replied:
Yeah, this is something the user define currently, so they could call whatever suits them

graham-chainlink committed:
Showcasing a sequence performing 3 operations

deploy link token (with no retry policy)
grant mint role
mint token (with modified input on retry)

graham-chainlink commented:
Closing this PR as it is only for demo purpose
